src/common/snippets/include/snippets/emitter.hpp

```diff
@@ -16,7 +16,13 @@ namespace snippets {
  * @interface RegType
  * @brief Register type of input and output operations
  */
-enum class RegType { gpr, vec, mask, undefined };
+enum class RegType {
+    gpr,
+    vec,
+    mask,
+    ignored,  // Used to mark regs that should be ignored by the code generation logic, as they are handled outside the snippets pipeline.
```

Comment:
Can we consider splitting register instance to properly support "ignored" values? Basically, ignored values are those which do not participate in register allocation. They are basically physical registers or immediate values or addresses.

We could consider having Operand class with inherited Register and Address (ImmValue) classes. And register could have types: physical and virtual (to be allocated).

Reply:
Discussed offline: it was decided to create a follow-up ticket for this activity. Within this PR, RegType renaming is done: `ignored` -> `address`

Reply:
Agreed to implement the suggestion within a separate activity